digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/SpeedLimitFiller.scala

```diff
@@ -291,7 +291,7 @@ object SpeedLimitFiller {
 
     val returnSpeedLimits = cleanSpeedLimitIds(resultingSpeedLimits, Seq())
     (returnSpeedLimits, changeSet.copy(droppedAssetIds = changeSet.droppedAssetIds ++ droppedIds,
-      adjustedMValues = changeSet.adjustedMValues ++ mValueAdjustments, adjustedSideCodes = changeSet.adjustedSideCodes ++ changedSideCodes))
+      adjustedMValues = changeSet.adjustedMValues ++ mValueAdjustments.filter(_.assetId > 0), adjustedSideCodes = changeSet.adjustedSideCodes ++ changedSideCodes))
```

Comment:
Don't return mValueAdjustments for speed limits that aren't yet saved to database.